node/src/protocol/message.rs

```diff
@@ -208,6 +209,20 @@ impl MessageHandler for RunningState {
         queue: &mut MpcMessageQueue,
     ) -> Result<(), MessageHandleError> {
         let mut triple_manager = self.triple_manager.write().await;
+
+        // remove the triple_id that has already failed from the triple_bins
+        queue
+            .triple_bins
+            .entry(self.epoch)
+            .or_default()
+            .retain(|id, _| {
+                let has_failed = triple_manager.failed_triples.contains_key(id);
+                if has_failed {
+                    triple_manager.failed_triples.insert(*id, Instant::now());
+                }
+                !has_failed
+            });
+
```

Comment:
Why we are not getting rid of the failed triple but adding failed triples?

Reply:
We are getting rid of the failed triples. This logic refreshes the timestamp of the failed triple, and then only retains the triples that have not failed.